### PR TITLE
Make some getters public, add various new mutable/immutable APIs and improve docs a bit

### PR DESCRIPTION
**Pull request by sdroege**

### CODE REVIEW

File: src/packet.rs [hunk 1]
```diff
@@ -235,7 +237,8 @@ impl<'a> RtpPacket<'a> {
         }
     }
 
-    fn payload_offset(&self) -> usize {
+    /// Returns the offset of the payload in this packet relative to the beginning of the packet.
+    pub fn payload_offset(&self) -> usize {
```
Comment (ystreet) [inline src/packet.rs:249]: Is there a particular reason you need this public?  Just trying to understand what this gives over accessing the payload/extension using the already provided API?

File: src/packet.rs [hunk 2]
```diff
@@ -154,7 +154,8 @@ impl<'a> RtpPacket<'a> {
         }
     }
 
-    fn extension_bit(&self) -> bool {
+    /// Returns whether the extension bit is set for this packet.
+    pub fn extension_bit(&self) -> bool {
```
Comment (ystreet) [inline src/packet.rs:159]: Same question, why do you need this over checking the return of `self.extension()`

File: src/packet.rs [hunk 3]
```diff
@@ -213,7 +214,8 @@ impl<'a> RtpPacket<'a> {
         Self::MIN_RTP_PACKET_LEN + (self.n_csrcs() as usize) * 4
     }
 
-    fn extension_len(&self) -> usize {
+    /// Returns the length of the extension data in this packet.
+    pub fn extension_len(&self) -> usize {
```
Comment (ystreet) [inline src/packet.rs:226]: Same question, why is this needed publicly.

File: src/packet.rs [hunk 4]
```diff
@@ -247,6 +247,14 @@ impl<'a> RtpPacket<'a> {
             }
     }
 
+    /// Returns the length of the payload in this packet without padding.
+    pub fn payload_len(&self) -> usize {
```
Comment (ystreet) [inline src/packet.rs:259]: Why is this needed over `self.payload().len()`?

**Review by ystreet — APPROVED** [review]

File: src/packet.rs (hunk 1 above)
Comment (sdroege) [reply src/packet.rs:249]: I want to keep a range around for the payload so I can easily access it at a later time without the lifetime constraints of only having a `&[u8]`. Currently I work around this with some pointer arithmetic to calculate the range from the two slices: https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/merge_requests/1424/diffs#0f3a605f32b7f9282e35bac4651a296b7d692f50_0_1193

File: src/packet.rs (hunk 2 above)
Comment (sdroege) [reply src/packet.rs:159]: Efficiency. I want to know beforehand if there's an extension, and at a later time (if there is one) I would access it. `extension()` is calculating a bit more than that, which probably gets optimized away though so it doesn't seem very important to keep this public.

Apart from that, also consistency. All other header flags have explicit accessors from what I can see.

File: src/packet.rs (hunk 3 above)
Comment (sdroege) [reply src/packet.rs:226]: Same answer: https://github.com/ystreet/rtp-types/pull/14#discussion_r1500301169

File: src/packet.rs (hunk 4 above)
Comment (sdroege) [reply src/packet.rs:259]: Consistency with `extension_len()` but otherwise not really needed.

File: src/packet.rs (hunk 2 above)
Comment (ystreet) [reply src/packet.rs:159]: The `padding_bit` does not have an explicit accessor and existence is currently only accessible via `padding()`.  I'm not opposed to this, but I think at least `marker()` would probably need to be renamed to `marker_bit()` (or be deprecated).

Comment (sdroege) [reply src/packet.rs:159]: Good question. I think that makes sense. Exposing them all as `..._bit()` and deprecating `marker()`.

Comment (sdroege) [reply src/packet.rs:159]: Question then is whether to also do `marker_bit()` for the mutable and builder structs. What do you think?

Comment (ystreet) [reply src/packet.rs:159]: I would lean towards yes for consistency.

Comment (sdroege) [reply src/packet.rs:159]: Ok, let's do that then. Thanks!

Comment (sdroege) [reply src/packet.rs:159]: @ystreet How about this?

Comment (ystreet) [reply src/packet.rs:159]: Looks good
